### PR TITLE
Rubocop -a

### LINKED ISSUE #8268: Rubocop cleanup
We all know that Spree codebase needs some cleaning and we've waited for the best moment to perform such cleanup. The moment is now. 3.4 release will be much smaller than releases before it so we can do a massive change like this. 

- [X] update [rubocop.yml](https://github.com/spree/spree/blob/master/.rubocop.yml)
- [ ] `rubocop -a api`
- [ ] `rubocop -a backend`
- [ ] `rubocop -a cmd`
- [ ] `rubocop -a core`
- [ ] `rubocop -a frontend`
- [ ] `rubocop -a guides`
- [ ] `rubocop -a sample`

### PR DESCRIPTION
**Pull request by bbonislawski**
fixes #8268

### CODE REVIEW

File: api/app/controllers/spree/api/base_controller.rb [hunk 1]
```diff
@@ -55,29 +55,29 @@ def authenticate_user
         return if @current_api_user
 
         if requires_authentication? && api_key.blank? && order_token.blank?
-          must_specify_api_key and return
+          must_specify_api_key && return
```
Comment (krzysiek1507) [inline api/app/controllers/spree/api/base_controller.rb]: it doesnt work :)

**Review by krzysiek1507 — DISMISSED** [review]

File: api/app/controllers/spree/api/base_controller.rb (hunk 1 above)
Comment (bbonislawski) [reply api/app/controllers/spree/api/base_controller.rb]: Not even one spec detects problem?

Comment (damianlegawiec) [reply api/app/controllers/spree/api/base_controller.rb]: @bbonislawski let's add regression specs + change rubocop config to omit the `&&` -> `and` rule

Comment (bbonislawski) [reply api/app/controllers/spree/api/base_controller.rb]: @damianlegawiec I excluded all controllers from AndOr cop. I'll try to add some regression specs

**Pull request merged**